packages/lesswrong/components/comments/CommentsNode.tsx

```diff
@@ -114,29 +114,26 @@ const CommentsNode = ({
   const currentUser = useCurrentUser();
   const { captureEvent } = useTracking()
   const scrollTargetRef = useRef<HTMLDivElement|null>(null);
-  const {hash: commentHash} = useSubscribedLocation();
-  const [collapsed, setCollapsed] = useState(!forceUnCollapsed && (comment.deleted || comment.baseScore < karmaCollapseThreshold));
-  const [truncatedState, setTruncated] = useState(!!startThreadTruncated);
 
-  // If the comment starts collapsed, check if the commentHash in the url matches any of this
-  // comment's children and uncollapse if so. This needs to be done inside a useEffect because
-  // the hash part of the url isn't sent to the server
-  useEffect(() => {
-    if (!collapsed) return;
+  const { linkedCommentId, scrollToCommentId } = useCommentLinkState();
+
+  const { lastCommentId, condensed, postPage, post, highlightDate, scrollOnExpand, forceSingleLine, forceNotSingleLine, expandOnlyCommentIds, noDOMId, onToggleCollapsed } = treeOptions;
```

Comment:
The changes in this file are mainly refactoring. The thing I was trying to achieve was:
- Handling scrolling in the `?commentId=...` case
- Making all the un-collapsing and un-truncating happen on the first render pass if in the `?commentId=...` case (previously with the hash it had to use a useEffect to avoid SSR mismatches)